src/celscript/legacy/cmdparser.cpp

```diff
@@ -9,1075 +9,1128 @@
+    Tokenizer tokenizer(&in);
+    int flags = 0;
+
+    Tokenizer::TokenType ttype = tokenizer.nextToken();
+    while (ttype != Tokenizer::TokenEnd)
     {
-        error("'{' expected at start of script.");
-        delete seq;
-        return nullptr;
+        if (auto tokenValue = tokenizer.getNameValue(); tokenValue.has_value())
+        {
+            if (compareIgnoringCase(*tokenValue, "all") == 0 && act==1)
+                cmd->flags.all = true;
+            else if (compareIgnoringCase(*tokenValue, "all") == 0 && act==0)
+                cmd->flags.none = true;
+            else
+                cmd->setValues(*tokenValue, act);
+
+            ttype = tokenizer.nextToken();
+            if (ttype == Tokenizer::TokenBar)
+                ttype = tokenizer.nextToken();
+        }
+        else
+        {
+            GetLogger()->error("Command Parser: error parsing render flags\n");
+            return 0;
+        }
     }
 
-    Tokenizer::TokenType ttype = tokenizer->nextToken();
-    while (ttype != Tokenizer::TokenEnd && ttype != Tokenizer::TokenEndGroup)
+    return flags;
+}
+
+
+int
+parseConstellationColor(CommandConstellationColor* cmd, const std::string &s, Eigen::Vector3d *col, int act)
+{
+    std::istringstream in(s);
+
+    Tokenizer tokenizer(&in);
+    int flags = 0;
+
+    if(!act)
+        cmd->unsetColor();
+    else
+        cmd->setColor(static_cast<float>(col->x()),
+                      static_cast<float>(col->y()),
+                      static_cast<float>(col->z()));
+
+    Tokenizer::TokenType ttype = tokenizer.nextToken();
+    while (ttype != Tokenizer::TokenEnd)
     {
-        tokenizer->pushBack();
-        Command* cmd = parseCommand();
-        if (cmd == nullptr)
+        if (auto tokenValue = tokenizer.getNameValue(); tokenValue.has_value())
         {
-            for_each(seq->begin(), seq->end(), [](Command* cmd) { delete cmd; });
-            delete seq;
-            return nullptr;
+            if (compareIgnoringCase(*tokenValue, "all") == 0 && act==1)
+                cmd->flags.all = true;
+            else if (compareIgnoringCase(*tokenValue, "all") == 0 && act==0)
+                cmd->flags.none = true;
+            else
+                cmd->setConstellations(*tokenValue);
+
+            ttype = tokenizer.nextToken();
+            if (ttype == Tokenizer::TokenBar)
+                ttype = tokenizer.nextToken();
         }
         else
         {
-            seq->push_back(cmd);
+            GetLogger()->error("Command Parser: error parsing render flags\n");
+            return 0;
         }
-
-        ttype = tokenizer->nextToken();
     }
 
-    if (ttype != Tokenizer::TokenEndGroup)
+    return flags;
+}
+
+
+ObserverFrame::CoordinateSystem parseCoordinateSystem(const std::string& name)
+{
+    using CoordinateSystemMap = std::map<std::string_view, ObserverFrame::CoordinateSystem, CompareIgnoringCasePredicate>;
+    static const CoordinateSystemMap* nameMap =
```

Comment:
i don't think that static map is a good replacement for the old code especially taking into account the tiny number of parameters.

old code should be faster and it doesn't need additional heap memory, it doesn't raise an issue with cleanup.

Reply:
What cleanup issue are you referring to here? Destructor doesn't get run since it is leaked (this is just a raw pointer, the `unique_ptr` gets immediately released), so shouldn't do anything to block process exit.

Reply:
i meant exactly that destructor is not running at all. just another issue reported by asan/valgrind. i think we should avoid such cases as much as we can or we'll be buried under asan reports.

Reply:
Not running destructors on static data and relying on the OS to clean up at process exit is a perfectly valid pattern though. If the tools can't be made aware of that, then they're bad tools.

Reply:
ok, but i still think that std::map should be used here. original code is much easier to understand

Reply:
Ok will remove it for some of these smaller ones, (the main if...else block that used to be in `parseCommand` I will leave as a map though, because that one was really unwieldy!).

Re: valgrind, maybe possible to set up a suppression file but not sure how portable that would be.